backend/llm_integration.py

```
@@ -32,18 +32,42 @@
+
+
+def _invoke_with_retry(client, messages, retries: int = 3):
+    """Invoke a LangChain client with exponential backoff retry."""
+    for attempt in range(retries):
+        try:
+            return client.invoke(messages)
+        except Exception as e:
+            if attempt == retries - 1:
+                raise
+            import time as _time
+            _time.sleep(2 ** attempt)
```

Ruff (0.15.20)

[warning] Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)